xbout/region.py

```diff
@@ -0,0 +1,744 @@
+from collections import OrderedDict
+
+import numpy as np
+import xarray as xr
+
+from .utils import _set_attrs_on_all_vars
+
+
+class Region:
+    """
+    Contains the global indices bounding a single topological region, i.e. a region with
+    logically rectangular contiguous data.
+
+    Also stores the names of any neighbouring regions.
+    """
+    def __init__(self, *, name, ds=None, xinner_ind=None, xouter_ind=None,
+                 ylower_ind=None, yupper_ind=None, connect_inner=None,
+                 connect_outer=None, connect_lower=None, connect_upper=None):
+        """
+        Parameters
+        ----------
+        name : str
+            Name of the region
+        ds : BoutDataset, optional
+            Dataset to get variables to calculate coordinates from
+        xinner_ind : int, optional
+            Global x-index of the inner points of this region
+        xouter_ind : int, optional
+            Global x-index of the points just beyond the outer edge of this region
+        ylower_ind : int, optional
+            Global y-index of the lower points of this region
+        yupper_ind : int, optional
+            Global y-index of the points just beyond the upper edge of this region
+        connect_inner : str, optional
+            The region inside this one in the x-direction
+        connect_outer : str, optional
+            The region outside this one in the x-direction
+        connect_lower : str, optional
+            The region below this one in the y-direction
+        connect_upper : str, optional
+            The region above this one in the y-direction
```

Comment:
Just curious, how consistent are we with `inner/outer` meaning `x`, and `lower/upper` meaning `y`?
Would it be better to be more explicit and include `x/y` in these names?

(also very minor, but the input args are `connect_*` and the member vars are `connection_*`; might be nice to make consistent and use the noun form?)

Reply:
Agreed, would be better with x and y in the names, an consistent `connection_*`.